data/filedefs/filetypes.json

```diff
@@ -0,0 +1,59 @@
+# For complete documentation of this file, please see Geany's main documentation
+[styling]
+# Edit these in the colorscheme .conf file instead
+default=default
+number=number_1
+string=string_1
+stringeol=string_eol
+propertyname=attribute
+escapesequence=string_2
+linecomment=comment_line
+blockcomment=comment
+operator=operator
+uri=string_1
+compactiri=string_1
```

Comment:
It might be better to map one consistent style to property names and compact IRIs (even though the latter are really a [component of JSON-LD][0]). NPM project files typically contain scripts with a single colon in their name. The JSON lexer sees them as compact IRIs, which is not what most users expect, and many of them will be editing NPM project files, like the one below (styled as per the filedef in git):


![npm-scripts-as-compact-iris](https://github.com/geany/geany/assets/59004801/9648b714-7246-4b12-bb31-353204abd6a9)

This "feature" has been the subject of a few bug reports already, e.g.,

- https://github.com/rizonesoft/Notepad3/issues/3899
- https://github.com/notepad-plus-plus/notepad-plus-plus/issues/11214#issuecomment-1040836577


[0]: https://www.w3.org/TR/json-ld/#dfn-compact-iri

Reply:
Did someone raise it on Lexilla?

Reply:
> It might be better to map one consistent style to property names and compact IRIs

Done now in the latest commit.